File: TransCoder_tokenized_test_set_functions/cpp/CHECK_GIVEN_ARRAY_CONTAINS_DUPLICATE_ELEMENTS_WITHIN_K_DISTANCE.cpp
bool checkDuplicatesWithinK ( int arr [ ], int n, int k ) {
  unordered_set < int > myset;
  for ( int i = 0;
  i < n;
  i ++ ) {
    if ( myset . find ( arr [ i ] ) != myset . end ( ) ) return true;
    myset . insert ( arr [ i ] );
    if ( i >= k ) myset . erase ( arr [ i - k ] );
  }
  return false;
}

File: TransCoder_tokenized_test_set_functions/python/CHECK_GIVEN_ARRAY_CONTAINS_DUPLICATE_ELEMENTS_WITHIN_K_DISTANCE.py
def checkDuplicatesWithinK ( arr, n, k ) :
    myset = [ ]
    for i in range ( n ) :
        if arr [ i ] in myset :
            return True
        myset . append ( arr [ i ] )
        if ( i >= k ) :
            myset . remove ( arr [ i - k ] )
    return False
